common-npm-packages/webdeployment-common/Tests/L1XmlVarSub.ts

```diff
@@ -1,29 +1,86 @@
-var xmlSubstitutionUtility = require('azure-pipelines-tasks-webdeployment-common/xmlvariablesubstitutionutility.js');
-var path = require('path');
-
-async function xmlVarSub() {
-    var tags = ["applicationSettings", "appSettings", "connectionStrings", "configSections"];
-    var configFiles = [path.join(__dirname, 'L1XmlVarSub/Web_test.config'), path.join(__dirname, 'L1XmlVarSub/Web_test.Debug.config')];
-    var variableMap = {
-        'conntype' : 'new_connType',
-        "MyDB": "TestDB",
-        'webpages:Version' : '1.1.7.3',
-        'xdt:Transform' : 'DelAttributes',
-        'xdt:Locator' : 'Match(tag)',
-        'DefaultConnection': "Url=https://primary;Database=db1;ApiKey=11111111-1111-1111-1111-111111111111;Failover = {Url:'https://secondary', ApiKey:'11111111-1111-1111-1111-111111111111'}",
-        'OtherDefaultConnection': 'connectionStringValue2',
-        'ParameterConnection': 'New_Connection_String From xml var subs',
-        'connectionString': 'replaced_value',
-        'invariantName': 'System.Data.SqlServer',
-        'blatvar': 'ApplicationSettingReplacedValue',
-        'log_level': 'error,warning',
-        'Email:ToOverride': ''
+import * as path from 'path';
+import * as assert from 'assert';
+import * as ltx from 'ltx';
+import * as fs from 'fs';
+
+import * as tl from 'azure-pipelines-task-lib';
+
+import { substituteXmlVariables } from '../xmlvariablesubstitutionutility';
+import { detectFileEncoding } from "../fileencoding";
+
+export function runL1XmlVarSubTests(this: Mocha.Suite): void {
+
+    this.timeout(60000);
```

Comment:
Do we really need this timeout, because from what I see - this test is synchronous.

Reply:
I think the `timeout` is applicable to sync tests too, isn't it? In this particular case it means that whole suite has to finish under one minute. Otherwise it fails.